src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -5,15 +5,23 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Linq;
+using Microsoft.AspNet.Mvc.DataAnnotations;
 using Microsoft.Extensions.Localization;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
 {
+    /// <summary>
+    /// Provides server-side validation of Models
+    /// </summary>
     public class DataAnnotationsModelValidator : IModelValidator
     {
         private IStringLocalizer _stringLocalizer;
+        private IModelMetadataProvider _modelMetadataProvider;
 
-        public DataAnnotationsModelValidator(ValidationAttribute attribute, IStringLocalizer stringLocalizer)
+        public DataAnnotationsModelValidator(
+            ValidationAttribute attribute,
+            IStringLocalizer stringLocalizer,
+            IModelMetadataProvider modelMetadataProvider)
```

Comment:
Place `modelMetadataProvider` before `stringLocalizer` and add a `null` check for it. Only :ok: for `stringLocalizer` to be `null`.